server/paymentProviders/thegivingblock/index.js

```diff
@@ -75,11 +106,11 @@ export const processOrder = async order => {
 
   // refresh credentials
   // TODO: we normally have to do it only every 2 hours but this handy for now
-  const { accessToken, refreshToken } = await refresh(account.data.refreshToken);
+  const { accessToken, refreshToken } = await refresh(account);
```

Comment:
refresh should take care of updating the account and return the updated account

Reply:
I've added the `account` update login within the `refresh` function. I think now it looks better. Let me know. 👍🏽